File: leetCode/fukun/lc_100.js
function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

const isSameTree = function (p, q) {
    if (!p && !q) return true;
    if (!q || !p || q.val !== p.val) return false;
    return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
};

let t1 = new TreeNode(1);
t1.left = new TreeNode(2);
t1.right = new TreeNode(3);

let t2 = new TreeNode(1);
t2.left = new TreeNode(2);
t2.right = new TreeNode(3);

console.log(isSameTree(t1, t2));

File: leetCode/lwc/0001-0100/lc_100.java
package solution2;

/**
 * @author lwc
 * @date 2019/10/27 11:32
 */
public class U100 {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p == null && q == null) return true;
        if(p == null || q==null || p.val != q.val) return false;
        return isSameTree(p.left,q.left) && isSameTree(p.right,q.right);
    }
}
